snorkel/classification/snorkel_classifier.py

```diff
@@ -330,10 +338,21 @@ def predict(
         prob_dict_list: Dict[str, List[torch.Tensor]] = defaultdict(list)
 
         for batch_num, (X_batch_dict, Y_batch_dict) in enumerate(dataloader):
-            prob_batch_dict = self._calculate_probs(X_batch_dict, Y_batch_dict.keys())
-            for task_name, Y in Y_batch_dict.items():
-                prob_dict_list[task_name].extend(prob_batch_dict[task_name])
-                gold_dict_list[task_name].extend(Y.cpu().numpy())
+            labels_to_tasks = self._get_labels_to_tasks(
+                label_names=Y_batch_dict.keys(), remap_labels=remap_labels
+            )
+
+            prob_batch_dict = self._calculate_probs(
+                X_batch_dict, labels_to_tasks.values()
+            )
+            for label_name in labels_to_tasks.keys():
```

Comment:
for label_name in labels_to_tasks does the same thing (.keys() is implicit)

Reply:
thanks!